### DISCUSSION TITLE
How to pass C# string[] to C char** or char[][]

### DISCUSSION THREAD

**Discussion by Sunguohua** [swig/swig#3200] [Q&A]
n C/C++ I have

typedef char TDAStringType[256];

struct CMarketReqMarketDataField
{
    TDACharType         MarketType;                     
    TDACharType         SubscMode;                      
    TDAIntType          MarketCount;                    
    TDAStringType       MarketTrcode[20];   
    TDAStringType       ErrorDescription;               
}
Look at the member MarketTrcode, It seems stock an array of string/char *, but I need to pass values to this struct in C/C++ through C# string[] frontend, How can I make it? using SWIG 4.0.2

**Comment by ojwb** [comment]
@Sunguohua Did you manage to resolve this?

I don't know the answer, but this seems a usage question rather than a bug in SWIG so moving to be a discussion - maybe if you still need an answer you'll get one there.
